GLOSSARY.md

```diff
@@ -0,0 +1,31 @@
+# Glossary
+
+This document gathers the canonical nomenclature of the Lightning concepts used by the LDK contributors. Note, they may diverge from the BOLT specs or the ones used by the wider Lightning community, the test of time has revealed that it can be hard to agree on terms employed. The following terms are employed with a best effort across the codebase aiming to reduce both code reviewers and library users confusion.
+
+### Channel Initiator
+
+In LDK context, the channel initiator is the entity who is taking the decision to open a channel. Finalization is relying on counterparty channel acceptance policy. Within current protocol version, the initiator has the burden to pay channel onchain fees.
+
+### Holder/Counterparty
+
+Inspired by financial cryptography, the holder is the entity beneficiary of a right. In LDK context, the holder is the entity who is served by the implementation in its channel operations. The counterparty is the entity participating in the channel operation as a peer, but of whom the interests are not of concern by the implementation.
```

Comment:
This begs the questions, what is being held by the holder? I feel using "financial cryptography" is trying to allude to something but it isn't clear as to what in the current phrasing.

Reply:
Right, a reworded a bit, "A holder is an owner of a given digital value. In Lightning, the holder is the entity owning a balance, of which the interests are served by the implementation during channel operations".

Reply:
Would you say "owner" and "holder" are synonymous in this context? I'm just a little confused still on the choice of the word "holder". Not to say that it isn't suitable. It just seems like it can be equally applied to "counterparty". For instance, isn't the counterparty also an owner of a balance, and thus also a holder?

Reply:
"holder" for us really means "local user/our machine/this process". Maybe s/is the entity owning a balance/is the entity operating the current node/?

Reply:
Yes good suggestion Matt, took it!

Reply:
I understand why this is a good definition. What I don't understand is why the term "holder" is a good term for what the definition is defining. 

Sorry about belaboring the point. Writing documentation like this is a good exercise because it helps us come up with suitable terminology. If we can't answer the question _what is held?_ or if the term is also applicable to the counterparty, then I'd argue that "holder" is not a suitable term.

If instead we can plug in "operator" (or some synonym) wherever we are using "holder" and if the relationship is between "operator" and "counterparty" is clear, then I'd argue that "operator" should be used throughout the codebase instead.

Reply:
Right, operator seems like a reasonable term, but either way renaming things would need to be a separate PR.